awx/main/dispatch/worker/base.py

```diff
@@ -156,6 +155,16 @@ def __init__(self, *args, **kwargs):
         # if no successful loops have ran since startup, then we should fail right away
         self.pg_is_down = True  # set so that we fail if we get database errors on startup
         self.pg_down_time = time.time() - self.pg_max_wait  # allow no grace period
+        self.last_cleanup = time.time()
+
+    def run_periodic_tasks(self):
+        self.record_statistics()  # maintains time buffer in method
+
+        if time.time() - self.last_cleanup > 20:  # same as cluster_node_heartbeat
```

Comment:
So I was wrong on this, the cluster_node_heatbeat period is 60, and also has an associated setting, `CLUSTER_NODE_HEARTBEAT_PERIOD`. So we should use that 60 second frequency. But you could also argue that we should use the setting itself... but I don't know if I like that, because this should be relatively independent.